sycl/test-e2e/Regression/static-buffer-dtor.cpp

```diff
@@ -21,9 +21,6 @@
 // UNSUPPORTED: windows && arch-intel_gpu_bmg_g21
 // UNSUPPORTED-TRACKER: https://github.com/intel/llvm/issues/17255
 
-// UNSUPPORTED: cuda
-// UNSUPPORTED-TRACKER: https://github.com/intel/llvm/issues/17450
```

Comment:
Note that this test is also failing on HIP, I suspect it might require the exact same patch for the HIP adapter

Reply:
I'm going to push the equivalent hip adapter patch but sadly it doesn't fix the fail. Even with everything on the UR side still alive during the static buffer destructor we get a segfault in libamdhip64.so.6 during this call https://github.com/intel/llvm/blob/sycl/unified-runtime/source/adapters/hip/device.hpp#L116. I tried sticking a `hipDeviceGet` call in just prior to that and it seems to work fine, but SetDevice still segfaults. I even tried adding a `hipInit` call there to no effect.

I can only assume that hip itself has some `static` or otherwise globally managed state that `hipSetDevice` relies on, as you can imagine there isn't much online about static lifetimes and hip objects